apps/settings/js/modules/addon_manager.js

```diff
@@ -43,7 +43,11 @@ define(function(require) {
      * @memberOf AddonManager.prototype
      */
     _bindEvents: function() {
-      AppsCache.addEventListener('oninstall', this._updateAddons.bind(this));
+      AppsCache.addEventListener('oninstall', (evt) => {
+        evt.application.addEventListener('downloadsuccess', (evt) => {
```

Comment:
please add some comment according to what fabrice provided, so we won't forget it after several months